recipes/fruit/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+from fnmatch import fnmatch
+import os
+import tarfile
+
+
+class FruitConan(ConanFile):
+    name = "fruit"
+    description = "C++ dependency injection framework"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/fruit"
+    license = "Apache-2.0"
+    topics = ("conan", "fruit", "injection")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "use_boost": [True, False],
+               "fPIC": [True, False]}
+    default_options = {"shared": False, "use_boost": True, "fPIC": True}
+    generators = "cmake", "cmake_find_package"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def build_requirements(self):
+        if self.options.use_boost:
+            self.build_requires("boost/1.72.0")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler = str(self.settings.compiler)
+        compiler_version = Version(self.settings.compiler.version.value)
+
+        minimal_version = {
+            "gcc": "5",
+            "clang": "3.5",
+            "apple-clang": "7.3",
+            "Visual Studio": "14"
+        }
+
+        if compiler in minimal_version and \
+           compiler_version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a compiler that supports"
+                                            " at least C++11. %s %s is not"
+                                            " supported." % (self.name, compiler, compiler_version))
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "11")
+
+    @property
+    def _extracted_dir(self):
+        return self.name + "-" + self.version
+
+    def _get_source(self):
+        filename = os.path.basename(self.conan_data["sources"][self.version]["url"])
+        tools.download(filename=filename, **self.conan_data["sources"][self.version])
+
+        with tarfile.TarFile.open(filename, 'r:*') as tarredgzippedFile:
```

Comment:
it might ]worth to submit an issue to the conan about it

Reply:
@SSE4 Yes. I created issue conan-io/conan#6611 and it was suggested to fix the archive file.

I created an issue google/fruit#107 but I don't know when the fix release will be.